components/mcb/Core/Src/mcb.c

```diff
@@ -189,7 +200,20 @@ void UpdateInputFlags(InputFlags * input_flags)
 {
 	input_flags->mech_brake_pressed = HAL_GPIO_ReadPin(BRK_IN_GPIO_Port, BRK_IN_Pin);
 	input_flags->regen_enabled = HAL_GPIO_ReadPin(REGEN_EN_GPIO_Port, REGEN_EN_Pin) && !input_flags->battery_regen_disabled;
-	input_flags->velocity_under_threshold = velocityOfCar < VELOCITY_THRESHOLD;
+
+	float* velocity = GET_CYCLIC_DATA(vehicle_velocity);
+	if (velocity != NULL)
+	{
+		input_flags->velocity_under_threshold = *velocity < VELOCITY_THRESHOLD;
+		input_flags->MDI_communication_fault = false;
+	}
+	else
+	{
+		// If NULL, then we haven't received a vehicle velocity msg from the MDI for MAX_CYCLE_TIME_VEHICLE_VELOCITY.
+		// Set 
+		input_flags->velocity_under_threshold = false;
```

Comment:
This might already be tested but I will ask to make sure: By setting velocity under threshold to false in the error state (MDI gives no message) this would mean that the firmware believes the car speed > 0.5 m/s. So if you wanted to go into park/transition to any state you would not be able to because velocity under threshold is false. This would be incorrect behavior in cases where we notice a fault and want to park the car or to reverse into a place where we can stop and diagnose the issue correct? Or is velocity_under_threshold being set elsewhere so its not permanently false?

Reply:
Since we don't have velocity measurements, we can't know for certain if the car is slow enough to change drive states.
In the future, we can add a new FAULT state that the MCB will enter when a fault is detected. Behavior in the FAULT state would need to be discussed.

Reply:
Since we already set the MDI no velocity fault to true could we use that as the criteria for the FAULT state? I agree it should be discussed further with some others. I think that since we simply cant know the velocity and the MCB cant override or do any motor controlling like the MDI does I would say ya we cant switch states and instead we just have to hold the brake.